### ISSUE TITLE
ScrapeConfig with BasicAuth add line-break to username/password fields

### ISSUE THREAD

**Issue by D13410N3** [prometheus-operator/prometheus-operator#6990]
### Is there an existing issue for this?

- [X] I have searched the existing issues

## Description
When I try to add ScrapeConfig resource with Basic Auth operator is adding line-breaks before username and password in Prometheus configuration. 

## Steps to Reproduce
1. Create ScrapeConfig with simple BasicAuth configuration:
```
spec:
  ### common SD or static configuration
  basicAuth:
    username:
      key: username
      name: foobar-basic-auth-username
    password:
      key: password
      name: foobar-basic-auth-password
```
2. Create secrets storing your username/password
```
---
apiVersion: v1
data:
  username: (base64)
kind: Secret
metadata:
  name: foobar-basic-auth-username
  namespace: monitoring
type: Opaque

---
apiVersion: v1
data:
  password: (base64)
kind: Secret
metadata:
  name:foobar-basic-auth-password
  namespace: monitoring
type: Opaque
```

3. Apply it and view runtime prometheus configuration. You will see this:
```
- job_name: scrapeConfig/...../...../
  basic_auth:
    username: |
      USERNAME
    password: |
      PASSWORD
```
As you can see operator added "|" in yaml fields

4. You're now getting 401 error, because there is "\n" after username & password
![image](https://github.com/user-attachments/assets/a0467d57-ae29-4cb3-87f1-f9a39c5024b5)

## Expected Result
No line-break added

## Actual Result
It is added and everything is broken

### Prometheus Operator Version

```shell
Image:      quay.io/prometheus-operator/prometheus-operator:v0.76.2
```
```


### Kubernetes Version

```yaml
clientVersion:
  buildDate: "2024-07-01T10:26:55Z"
  compiler: gc
  gitCommit: "0000000000000000000000000000000000000000"
  gitTreeState: archive
  gitVersion: v1.27.15
  goVersion: go1.20.5
  major: "1"
  minor: "27"
  platform: linux/amd64
kustomizeVersion: v5.0.1
serverVersion:
  buildDate: "2024-07-01T10:25:14Z"
  compiler: gc
  gitCommit: "0000000000000000000000000000000000000000"
  gitTreeState: archive
  gitVersion: v1.27.15
  goVersion: go1.20.5
  major: "1"
  minor: "27"
  platform: linux/amd64
```

### Kubernetes Cluster Type

Other (please comment)

### How did you deploy Prometheus-Operator?

prometheus-operator/kube-prometheus

### prometheus-operator log output

```shell
no log
```

**Comment by simonpasquier** [comment]
Can you double check that the secret data doesn't contain `\n` characters? I see no way that the operator would add them.

**Comment by D13410N3** [comment]
@simonpasquier Hello. I don't see any characters in secret
```
apiVersion: v1
data:
  password: <base64>
kind: Secret
metadata:
  annotations:
  labels:
    app.kubernetes.io/instance: monitoring-stack
  name: <name>
  namespace: <namespace>
type: Opaque
```

The same for username. 

I've tried encode/decode my values again - and they are OK, without `\n`

**Comment by D13410N3** [comment]
Resolved.
https://github.com/prometheus-operator/prometheus-operator/issues/1568#issuecomment-402487974

TL;DR: 
`echo 'foobar' | base64` - will transform string to `'foobar\n'` first and then encode it
Always use `echo -n 'foobar' | base64` to prevent adding line-break
